poetry/utils/env.py

```diff
@@ -417,7 +417,7 @@ def remove(self, python):  # type: (str) -> Env
         envs_file = TomlFile(venv_path / self.ENVS_FILE)
         base_env_name = self.generate_env_name(self._poetry.package.name, str(cwd))
 
-        if python.startswith(base_env_name):
+        if python.startswith(base_env_name) or python == ".venv":
```

Comment:
We shouldn't really do this if the environment was not created by us. There needs to be some flag somewhere (maybe even a marker file) in the env directory to determine if we created it.

Reply:
How about checking `self._poetry.config.get("virtualenvs.in-project")` is true, similar to what's done when listing the envs?

Reply:
The issue is that poetry supports existing environments. So, in cases where, lets say, someone needs some wierd system lib available etc., they might create `.venv` manually and also have `virtualenvs.in-project` set to `true`. In this scenario, there is no guarentee that we created it. Obviously other simpler scenarios exists too. Just picked an obtuse one.

Reply:
Does a marker file / flag like you say exist somewhere already or will that need to be added when Poetry creates a `.venv`?

Reply:
At the moment no. If we need to support this cleanly we should add one and add a good message if one is not found when user tries to delete this. 

Virtualenv already creates a `pyvenv.cfg`, either relying on that or placing a `poetry.toml` file in the environment root maybe sufficient. Not sure what is the better approach here at the moment.

Reply:
The latter affords expansion later.

Reply:
So for a workaround (other than `rm` as noted in the initial issue) until #2924 (presumably?) changes the approach for a `.venv` would the following be sufficient?

1. Check `poetry.toml` exists in the project root.
2. Check `poetry.toml` contains `in-project = true` in the appropriate place.

Would a situation exist where a user has set locally `in-project = true` but has manually created `.venv` _and_ doesn't want to remove it when running `poetry env remove .venv`?